src/row.rs

```diff
@@ -1,25 +1,54 @@
-use std::{collections::HashMap, sync::Arc};
+use std::{collections::HashMap, ops::Deref, sync::Arc};
 
 use chrono::{DateTime, Days, NaiveDate, NaiveDateTime};
 
 use crate::{Error, Result};
 
+#[derive(Debug)]
+pub struct SnowflakeColumn {
+    pub name: String,
+    pub column_type: SnowflakeColumnType,
+}
+
+#[derive(Debug, Clone)]
+pub struct SnowflakeColumnType {
+    /// The index of the column in the row
+    pub index: usize,
```

Comment:
`SnowflakeColumn` might be a better place to put `index` than `SnowflakeColumnType` since it's not related to the type.